src/Microsoft.DotNet.VersionTools/Dependencies/Submodule/IndicatorPackageSubmoduleUpdater.cs

```diff
@@ -0,0 +1,96 @@
+    {
+        /// <summary>
+        /// The NuGet v2 base url to use to download the indicator package, without a trailing '/'.
+        /// For example, https://dotnet.myget.org/F/dotnet-core/api/v2/package.
+        /// </summary>
+        public string PackageDownloadBaseUrl { get; set; }
+
+        public string IndicatorPackageId { get; }
+
+        public IndicatorPackageSubmoduleUpdater(string indicatorPackageId)
+        {
+            if (indicatorPackageId == null)
+            {
+                throw new ArgumentNullException(nameof(indicatorPackageId), "An indicator package must be specified.");
+            }
+            IndicatorPackageId = indicatorPackageId;
+        }
+
+        protected override string GetDesiredCommitHash(
+            IEnumerable<DependencyBuildInfo> dependencyBuildInfos,
+            out IEnumerable<DependencyBuildInfo> usedBuildInfos)
+        {
+            foreach (var info in dependencyBuildInfos)
+            {
+                PackageIdentity package = info.Packages
+                    .FirstOrDefault(p => p.Id == IndicatorPackageId);
+
+                if (package == null)
+                {
+                    continue;
+                }
+
+                using (ZipArchive archive = DownloadPackageAsync(info, package).Result)
+                {
+                    Stream versionTxt = archive.GetEntry("version.txt")?.Open();
+                    if (versionTxt == null)
+                    {
+                        Trace.TraceWarning(
+                            $"Downloaded '{package}' in '{info.BuildInfo.Name}' " +
+                            $"to upgrade '{Path}', but it had no version.txt file. Skipping.");
+                        continue;
+                    }
+
+                    string packageCommitHash = new StreamReader(versionTxt).ReadLine();
+                    Trace.TraceInformation($"Found commit '{packageCommitHash}' in versions.txt.");
+
+                    usedBuildInfos = new[] { info };
+                    return packageCommitHash;
+                }
+            }
+
+            Trace.TraceError($"Failed to find '{IndicatorPackageId}' specifying a commit in any build-info.");
+            usedBuildInfos = Enumerable.Empty<DependencyBuildInfo>();
+            return null;
+        }
+
+        protected async Task<ZipArchive> DownloadPackageAsync(DependencyBuildInfo info, PackageIdentity package)
+        {
+            if (PackageDownloadBaseUrl == null)
+            {
+                throw new NotSupportedException(
+                    $"A {nameof(PackageDownloadBaseUrl)} must be configured, " +
+                    "as build-infos do not have package feed details.");
+            }
+
+            string downloadUrl = $"{PackageDownloadBaseUrl}/package/{package.Id}/{package.Version}";
+            Trace.TraceInformation($"Downloading '{package}' from '{downloadUrl}'");
+
+            var client = new HttpClient();
```

Comment:
Can we put this in a static so we can re-use them?  It's safe to have multiple operations in flight on the same HttpClient object.